### DISCUSSION TITLE
Page automatically reloads every time code is saved when using NextIntlClientProvider of next-intl

### DISCUSSION THREAD

**Discussion by baoanh177** [community/community#147742] [Programming Help]
### Body

I'm setting up `next-intl` for translation in my Nextjs App Router project and I'm having a problem that every time I save the code, even the code has no changes, my page running localhost gets reloaded.

I tried and found that the problem is caused by `NextIntlClientProvider`. When I removed it, the above situation no longer happened on my website. But I still don't know how to solve it, does anyone have the same problem as me?

This is `src/app/layout.tsx`
![image](https://github.com/user-attachments/assets/84c7ddbd-970a-49e7-ac6c-ee4c61428368)
This is next.config.mjs
![image](https://github.com/user-attachments/assets/bd563f89-c2f2-4d3e-8973-8cdbc0c023dd)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by BryanLomerio** [comment]
Make sure you’re using compatible versions of next and next-intl. If unsure, check your package.json.

Add Hot Reload Fix: In your next.config.js add the following to prevent unnecessary reloads:

module.exports = {
  webpackDevMiddleware: (config) => {
    config.watchOptions = {
      poll: 1000,
      aggregateTimeout: 300,
    };
    return config;
  },
};

Simplify NextIntlClientProvider Setup: Ensure you're using the NextIntlClientProvider correctly. For example:


import { NextIntlClientProvider } from 'next-intl';

export default function Layout({ children }) {
  const messages = await import(`./messages/${locale}.json`);

  return (
    <NextIntlClientProvider messages={messages}>
      {children}
    </NextIntlClientProvider>
  );
}
Ensure Proper i18n Setup: Check that your next.config.mjs has the correct i18n configuration:


export default {
  i18n: {
    locales: ['en', 'es'], // Add your locales
    defaultLocale: 'en',
  },
};
